cpp/src/strings/convert/convert_hex.cu

```diff
@@ -141,6 +142,44 @@ std::unique_ptr<column> hex_to_integers(
   return results;
 }
 
+std::unique_ptr<column> is_hex(strings_column_view const& strings,
+                               cudaStream_t stream,
+                               rmm::mr::device_memory_resource* mr)
+{
+  auto strings_column = column_device_view::create(strings.parent(), stream);
+  auto d_column       = *strings_column;
+  // create output column
+  auto results   = make_numeric_column(data_type{type_id::BOOL8},
+                                     strings.size(),
+                                     copy_bitmask(strings.parent(), stream, mr),
+                                     strings.null_count(),
+                                     stream,
+                                     mr);
+  auto d_results = results->mutable_view().data<bool>();
+  thrust::transform(rmm::exec_policy(stream)->on(stream),
+                    thrust::make_counting_iterator<size_type>(0),
+                    thrust::make_counting_iterator<size_type>(strings.size()),
+                    d_results,
+                    [d_column] __device__(size_type idx) {
+                      if (d_column.is_null(idx)) return false;
+                      auto const d_str = d_column.element<string_view>(idx);
+                      if (d_str.empty()) return false;
+                      auto const starts_with_0x = [](auto const& sv) {
+                        return sv.length() > 1 && (sv.substr(0, 2) == string_view("0x", 2) ||
```

Comment:
should these `string_view`'s be initialized outside of the lambda and reused? wondering about perf. Probably won't make a difference.

Reply:
Creating these outside the `transform` lambda would require extra cudaMemcpys. I did a quick profile using 10K strings and the above code is faster without the extra cudaMemcpys even though the kernel uses more instructions to setup the compare.